### PR TITLE
Update Locking docs to refer `server-id`

### PR DESCRIPTION
**Pull request by tigrato**
PR #27018 extended support for locking any type of Teleport instances besides `nodes`. 

This PR updates the documentation to be aligned with the new `--server-id` flag.

### CODE REVIEW

File: docs/pages/access-controls/guides/locking.mdx [hunk 1]
```diff
@@ -30,7 +30,7 @@ A lock can target the following objects or attributes:
 - a Teleport [trusted device](device-trust.mdx#optional-locking-a-device) by the device ID
 - an MFA device by the device's UUID
 - an OS/UNIX login
-- a Teleport Node by the Node's UUID (effectively unregistering it from the
+- a Teleport agent by the Agent's Server UUID (effectively unregistering it from the
```
Comment (ptgott) [inline docs/pages/access-controls/guides/locking.mdx:33]:
```suggestion
- a Teleport agent by the agent's server UUID (effectively unregistering it from the
```

File: docs/pages/access-controls/guides/locking.mdx [hunk 2]
```diff
@@ -74,10 +74,10 @@ with one of the following options:
   # Created a lock with name "d6c06a18-e147-4232-9dfe-6f83a28d5850".
   ```
   </TabItem>
-  <TabItem label="Node">
-  All connections to the specified node will be locked.
+  <TabItem label="Agent">
+  All connections to the specified agent will be locked and the agent will be excluded from the Teleport cluster.
   ```code
-  $ tctl lock --node=363256df-f78a-4d99-803c-bae19da9ede4 --message="The node is under investigation." --ttl=10h
+  $ tctl lock --server-id=363256df-f78a-4d99-803c-bae19da9ede4 --message="The server running Kubernetes Service and Database service is under investigation." --ttl=10h
```
Comment (ptgott) [inline docs/pages/access-controls/guides/locking.mdx:80]:
```suggestion
  $ tctl lock --server-id=363256df-f78a-4d99-803c-bae19da9ede4 --message="The server running the Kubernetes Service and Database Service is under investigation." --ttl=10h
```

**Review by ptgott — APPROVED** [review]
Approved with minor suggestions

**Comment by ptgott** [comment]
@tigrato Do we need to backport this?

**Comment by tigrato** [comment]
> @tigrato Do we need to backport this?

Yes, I will backport it to v11-v13

**Review by zmb3 — APPROVED** [review]

**Pull request merged**